src/queueUntilType.js

```diff
@@ -0,0 +1,35 @@
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+export function queueUntilType(expectedType) {
+  return this.lift(function(source) {
+    let queue = [];
+    let hasSeenType = false;
+    const nullifyQueue = () => { queue = null; };
+
+    const outer = {
+      notifyNext: (_, action) => this.next(action),
+      notifyComplete: nullifyQueue
+    };
+    return source.subscribe({
+      next: action => {
+        // It's important to check this first because of possible re-entry
+        // before we've finished flushing the queue.
+        if (hasSeenType) {
+          return this.next(action);
+        }
+
+        if (action && action.type === expectedType) {
+          hasSeenType = true;
+          this.add(subscribeToResult(outer, queue));
+          // no longer needed, free up memory
+          nullifyQueue();
+        } else {
+          queue.push(action);
+        }
+      },
+      error: e => this.error(e),
+      complete: () => this.complete(),
+      unsubscribe: nullifyQueue
```

Comment:
we're digging into this btw.. some interesting examples so far:

* called on error https://jsbin.com/zudivez/edit?js,console (edited)
* called on complete https://jsbin.com/nazixix/edit?js,console (edited)
* NOT called on unsubscribe https://jsbin.com/lerilub/edit?js,console (edited)

(but `finally()` is called for all three)

Current thinking is either this shouldn't be supported at all or should have the same behavior as `finally()` and thus is a bug. Not sure which yet.